src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedPhysicalPlanPrinterImpl2.java

```diff
@@ -0,0 +1,149 @@
+package se.liu.ida.hefquin.engine.queryplan.utils;
+
+import java.io.PrintStream;
+
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpBGPAdd;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalPlanVisitor;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalPlanWalker;
+import se.liu.ida.hefquin.engine.queryplan.physical.impl.*;
+import se.liu.ida.hefquin.engine.utils.IndentingPrintStream;
+
+public class TextBasedPhysicalPlanPrinterImpl2 implements PhysicalPlanPrinter
+{
+	// The string represents '|   '.
+	private static String levelIndentBase = "\u2502   ";
+	// The string represents '├── '.
+	private static String nonLastChildIndentBase = "\u251C\u2500\u2500 ";
+	// The string represents '└── '.
+	private static String lastChildIndentBase = "\u2514\u2500\u2500 ";
+	private static String spaceBase = "    ";
```

Comment:
Assuming these things are the same as in `TextBasedLogicalPlanPrinterImpl2`, I propose to create a base class for both of the cases and move these common things to that base class (in which case you will need to change `private` to `protected`). The base class may be called `BaseForTextBasedPlanPrinters`.